modules/dev-utils/src/test/java/org/apache/ignite/development/utils/IgniteWalConverterTest.java

```diff
@@ -0,0 +1,381 @@
+            while (!find) {
+                int recordTypeIndex = raf.read();
+                if (recordTypeIndex > 0) {
+                    recordTypeIndex--;
+                    final long idx = raf.readLong();
+                    final int fileOff = Integer.reverseBytes(raf.readInt());
+                    final int len = Integer.reverseBytes(raf.readInt());
+                    if (recordTypeIndex == WALRecord.RecordType.DATA_RECORD.index()) {
+                        int i = 0;
+                        int b;
+                        while (!find && (b = raf.read()) >= 0) {
+                            if (findByte[i] == b) {
+                                i++;
+                                if (i == findByte.length) {
+                                    find = true;
+                                }
+                            }
+                            else {
+                                i = 0;
+                            }
+                        }
+                        if (find) {
+                            raf.seek(raf.getFilePointer() - 1);
+                            raf.write(' ');
+                        }
+                    }
+                    raf.seek(fileOff + len);
+                }
+            }
+        }
+
+        final ByteArrayOutputStream outByte = new ByteArrayOutputStream();
+        final PrintStream out = new PrintStream(outByte);
+        IgniteWalConverter.convert(
+            out,
+            DataStorageConfiguration.DFLT_PAGE_SIZE,
+            walDir,
+            U.resolveWorkDirectory(U.defaultWorkDirectory(), "wal_archive", false),
+            new File(U.resolveWorkDirectory(U.defaultWorkDirectory(), "binary_meta", false), nodeFolder),
+            U.resolveWorkDirectory(U.defaultWorkDirectory(), "marshaller", false),
+            false
+        );
+
+        final String result = outByte.toString();
+        int index = 0;
+        int countErrorRead = 0;
+        for (Person person : list) {
+            boolean find = false;
+            index = result.indexOf("DataRecord", index);
+            if (index > 0) {
+                index = result.indexOf("PersonKey", index + 10);
+                if (index > 0) {
+                    index = result.indexOf("id=" + person.getId(), index + 9);
+                    if (index > 0) {
+                        index = result.indexOf(person.getClass().getSimpleName(), index + 4);
+                        if (index > 0) {
+                            index = result.indexOf("id=" + person.getId(), index + person.getClass().getSimpleName().length());
+                            if (index > 0) {
+                                index = result.indexOf("name=" + person.getName(), index + 4);
+                                find = index > 0;
+                            }
+                        }
+                    }
+                }
+            }
+            if (!find) {
+                countErrorRead++;
```

Comment:
System.setProperty(IgniteSystemProperties.IGNITE_PDS_SKIP_CRC, "true");
RecordV1Serializer.skipCrc = true;
Disables the crc check.
The test is made so that spoils value in the interior DataRecord and does not find a match to the added data (countErrorRead == 1)

Reply:
Yes, I got it, but I didn't see any difference in output between correct and incorrect records. There should be a way to distinguish them.